### PR TITLE
MV-113 Add socket connection params

### PR DESCRIPTION
**Pull request by whoiswentz**
This PR introduces the possibility to send params to phoenix transport when connecting to it

### CODE REVIEW

File: app/src/main/java/com/dscout/membranevideoroomdemo/MainActivity.kt [hunk 1]
```diff
@@ -66,11 +67,12 @@ class MainActivity : AppCompatActivity() {
         showSystemUi = true,
     )
     @Composable
-    fun MainContent(onConnect: (String, String) -> Unit = { _, _ -> }) {
+    fun MainContent(onConnect: (String, String, Params?) -> Unit = { _, _, _ -> }) {
 
         val scrollableState = rememberScrollState()
         val roomName = remember { mutableStateOf(TextFieldValue("room")) }
         val displayName = remember { mutableStateOf(TextFieldValue("Android User")) }
+        val params = mapOf<String, Any>("token" to "mocktoken")
```
Comment (graszka22) [inline app/src/main/java/com/dscout/membranevideoroomdemo/MainActivity.kt:75]: put the mocked params in RoomViewModel, that way you don't have to modify MainActivity and RoomActivity
concerns like auth tokens should be in ViewModels

File: app/src/main/java/com/dscout/membranevideoroomdemo/utils/types.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.dscout.membranevideoroomdemo.utils
+
+public typealias Params = Map<String, Any>
```
Comment (graszka22) [inline app/src/main/java/com/dscout/membranevideoroomdemo/utils/types.kt:3]: this is not used anywhere

**Review by graszka22 — COMMENTED** [review]
thanks, I tested it and it's working, just a couple of comments to make the code simpler

File: app/src/main/java/com/dscout/membranevideoroomdemo/MainActivity.kt (hunk 1 above)
Comment (whoiswentz) [reply app/src/main/java/com/dscout/membranevideoroomdemo/MainActivity.kt]: I made a change, can you check and let me know if this is what you meant

File: app/src/main/java/com/dscout/membranevideoroomdemo/utils/types.kt (hunk 2 above)
Comment (whoiswentz) [reply app/src/main/java/com/dscout/membranevideoroomdemo/utils/types.kt]: yeah, sorry, removed

**Review by graszka22 — COMMENTED** [review]
👍 thanks a lot!

**Review by graszka22 — APPROVED** [review]

File: MembraneRTC/src/main/java/org/membraneframework/rtc/transport/PhoenixTransport.kt [hunk 3]
```diff
@@ -7,14 +7,16 @@ import org.membraneframework.rtc.events.ReceivableEvent
...
 import timber.log.Timber
 
 public class PhoenixTransport constructor(
-    val url: String,
-    val topic: String,
-    private val ioDispatcher: CoroutineDispatcher
+    private val url: String,
+    private val topic: String,
+    private val ioDispatcher: CoroutineDispatcher,
+    private val params: SocketConnectionParams? = null
```
Comment (graszka22) [inline MembraneRTC/src/main/java/org/membraneframework/rtc/transport/PhoenixTransport.kt:19]: One more thing, can we have `emptyMap()` instead `null`? We don't want to break backward compatibility if it's not necessary

Comment (whoiswentz) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/transport/PhoenixTransport.kt]: done!

**Review by graszka22 — APPROVED** [review]

**Pull request merged**